src/php/ext/grpc/config.m4

```diff
@@ -7,35 +7,51 @@ PHP_ARG_ENABLE(coverage, whether to include code coverage symbols,
 PHP_ARG_ENABLE(tests, whether to compile helper methods for tests,
 [  --enable-tests          Enable tests methods], no, no)
 
+PHP_ARG_WITH(grpc-dir,	for grpc,
+[	--with-grpc-dir[=DIR]	 Set the path to grpc install prefix.], yes)
+
 dnl Check whether to enable tests
 if test "$PHP_TESTS" != "no"; then
   CPPFLAGS="$CPPFLAGS -DGRPC_PHP_DEBUG"
 fi
 
 if test "$PHP_GRPC" != "no"; then
-  dnl Write more examples of tests here...
-
-  dnl # --with-grpc -> check with-path
-  SEARCH_PATH="/usr/local /usr"     # you might want to change this
-  SEARCH_FOR="include/grpc/grpc.h"  # you most likely want to change this
-  if test -r $PHP_GRPC/$SEARCH_FOR; then # path given as parameter
-    GRPC_DIR=$PHP_GRPC
-  else # search default path list
+  AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
+
+  if test "$PHP_GRPC_DIR" = "yes" -a -x $PKG_CONFIG; then
+    AC_MSG_CHECKING([for grpc using pkg-config])
+
+    if ! $PKG_CONFIG --exists grpc ; then
+			AC_MSG_ERROR([grpc not found])
+		fi
+
+    GRPC_VERSION=`$PKG_CONFIG grpc --modversion`
+		AC_MSG_RESULT([found version $GRPC_VERSION])
+
+    PHP_GRPC_LIBS=`$PKG_CONFIG grpc --libs`
+		PHP_GRPC_INCS=`$PKG_CONFIG grpc --cflags`
+
+    PHP_EVAL_LIBLINE($PHP_GRPC_LIBS, AMQP_SHARED_LIBADD)
+    PHP_EVAL_INCLINE($PHP_GRPC_INCS)
+  else
```

Comment:
Why not use PHP pkg-config macros?

https://github.com/php/php-src/commit/cc9c5d80570c0a3184017a921099b674f5e119e9